setup.cfg

```diff
@@ -77,6 +77,7 @@ dev =
     mypy
     tox
     numpydoc
+    ruff
```

Comment:
So for ruff is not run by default, right? I expect a lot of formatting changes and complaint about future deprecations when running ruff on the existing code. (We have to do it a some point ...)

Reply:
I'm running ruff on the repo by default, it doesn't do any formatting changes anymore, they have been done already. They were minor (it is pretty close to black's style).

With the default ruleset, also `ruff check` is clean. I haven't checked with the CR ruleset.

Reply:
Maybe this wasn't clear: We are running ruff already in the pre-commit (so also on github in the checks) in both checking and formatting mode. We have a reduced set of ruff check rules active compared to other CR projects. If you added the documentation rules and some other stuff, we would get lots of warnings, some of them actually things we should fix, some of them just stupid. At some point I could look through this, but I don't think it is pressing.